config/configgrpc/configgrpc.go

```diff
@@ -260,10 +261,13 @@ func (gcs *GRPCClientSettings) toDialOptions(host component.Host, settings compo
 
 	otelOpts := []otelgrpc.Option{
 		otelgrpc.WithTracerProvider(settings.TracerProvider),
-		otelgrpc.WithMeterProvider(settings.MeterProvider),
 		otelgrpc.WithPropagators(otel.GetTextMapPropagator()),
 	}
 
+	if obsreportconfig.EnableHighCardinalityMetricsfeatureGate.IsEnabled() {
+		otelOpts = append(otelOpts, otelgrpc.WithMeterProvider(settings.MeterProvider))
```

Comment:
Will adding a view to filter out this metric still allocate memory for all of the potential attribute values? Are there any known performance hits we would take by adding a view?

Reply:
My understanding is that memory would not be allocated and that the performance hit would be negligible (it just means testing if the view matches against instruments at creation time). More concretely, adding a Drop view means that 
- if the instrument is synchronous, the [`aggregators` array on the instrument](https://github.com/open-telemetry/opentelemetry-go/blob/d1959c9239ae1559e828e54cfd05c7bd4adb9c56/sdk/metric/instrument.go#L207) would be `nil`, which would remove any work associated with the metric (it will not be aggregated, nor will it show up in the Prometheus scrape). I guess after looking at the code my question would be if [creating the new set](https://github.com/open-telemetry/opentelemetry-go/blob/d1959c9239ae1559e828e54cfd05c7bd4adb9c56/sdk/metric/instrument.go#L206) allocates any memory (I don't think so from a brief look at the implementation but if it did it could be solved in OTel Go side),
- if the instrument is asynchronous, the callback would not be registered at all, so it would never be called

I guess @Aneurysm9 can settle this more definitively than I can, but in summary I think it should be fine :)

Reply:
I agree w/ @mx-psi, I'd expect this to be a view. In a future where views are configurable by end users, this is likely how we would expect it to be filtered out.

Reply:
+1 to using a view.